pygount/write.py

```diff
@@ -128,154 +131,51 @@ def close(self):
-        percentage_width = 4 + digits_after_dot  # To represent "100." we need 4 characters.
-
-        summary_heading = (
-            "{0:^{max_language_width}s}  "
-            "{1:^{max_file_count_width}s}  "
-            "{2:^{percentage_width}s}  "
-            "{3:^{max_code_width}s}  "
-            "{2:^{percentage_width}s}  "
-            "{4:^{max_documentation_width}s}  "
-            "{2:^{percentage_width}s}"
-        ).format(
-            SummaryWriter._LANGUAGE_HEADING,
-            SummaryWriter._FILE_COUNT_HEADING,
-            "%",
-            SummaryWriter._CODE_HEADING,
-            SummaryWriter._DOCUMENTATION_HEADING,
-            max_code_width=max_code_width,
-            max_file_count_width=max_file_count_width,
-            max_documentation_width=max_documentation_width,
-            max_language_width=max_language_width,
-            percentage_width=percentage_width,
-        )
-        self._target_stream.write(summary_heading + os.linesep)
-        separator_line = "  ".join(
-            [
-                "-" * width
-                for width in (
-                    max_language_width,
-                    max_file_count_width,
-                    percentage_width,
-                    max_code_width,
-                    percentage_width,
-                    max_documentation_width,
-                    percentage_width,
-                )
-            ]
-        )
-        self._target_stream.write(separator_line + os.linesep)
-        language_line_template = (
-            "{0:{max_language_width}s}  "
-            "{1:>{max_file_count_width}d}  "
-            "{2:>{percentage_width}.0{digits_after_dot}f}  "
-            "{3:>{max_code_width}d}  "
-            "{4:>{percentage_width}.0{digits_after_dot}f}  "
-            "{5:>{max_documentation_width}d}  "
-            "{6:>{percentage_width}.0{digits_after_dot}f}"
-        )
-        for language_summary in sorted(self.project_summary.language_to_language_summary_map.values(), reverse=True):
-            code_count = language_summary.code_count
-            code_percentage = (
-                code_count / self.project_summary.total_code_count * 100
-                if self.project_summary.total_code_count != 0
-                else 0.0
-            )
-            file_count = language_summary.file_count
-            assert (
-                self.project_summary.total_file_count != 0
-            ), "if there is at least 1 language summary, there must be a file count too"
-            file_percentage = file_count / self.project_summary.total_file_count * 100
-            documentation_percentage = (
-                language_summary.documentation_count / self.project_summary.total_documentation_count * 100
-                if self.project_summary.total_documentation_count != 0
-                else 0.0
-            )
-            line_to_write = language_line_template.format(
+        table = Table()
+
+        for col in self._TEXT_COLUMNS:
```

Comment:
I think this would be easier to read and understand:
```python
_COLUMNS_WITH_JUSTIFY = (
   ("Language", "left"),
   ("Files", "right"),
   ("Blank", "right"),
   ("Comment", "right"),
   ("Code", "right"),
)
...

table = Table()
for column, justify in _COLUMNS_WITH_JUSTIFY:
   table.add(column, justify)
```
The code above also spells out the unnecessary abbreviation `col` and reverts the column headings to consistently start with an upper case letter.

Reply:
Cleaned up.